src/tree.service.ts

```diff
@@ -24,6 +25,8 @@ export class TreeService {
   public nodeExpanded$: Subject<NodeExpandedEvent> = new Subject<NodeExpandedEvent>();
   public nodeCollapsed$: Subject<NodeCollapsedEvent> = new Subject<NodeCollapsedEvent>();
 
+  private controllers: Map<string | number, TreeController> = new Map();
```

Comment:
It doesn't mater if you rename a node, because you change only the name of the node, not it's id.
About the destroy I will add it :)